src/data/resolvers/motions.ts

```diff
@@ -1310,20 +1310,21 @@ export const motionsResolvers = ({
           apolloClient,
         );
 
-        const safeTxMetadata = JSON.parse(
-          (await getColonyMetadataIPFS(annotation?.values?.metadata || '')) ||
-            '[]',
-        );
+        if (annotation) {
+          const safeTxMetadata = await getColonyMetadataIPFS(
```

Comment:
I'm a bit surprised with getColonyMetadataIPFS It doesnt really make use of ColonyMetadataEventParser.
The func is returning (promise)any, but could be (type) ColonyMetadata as the name suggests.

Also, inside getColonyMetadataIPFS it doesnt handle different version of metadataVersion. I guess in theory we would not expect any safe data to be encoded in V1, everything should now be V2, but its good to have the framework so we can easily support V3.

Not using ColonyMetadataEventParser to parse V2 data means that yup validation is not be used.

What happens if the data is malformed?

If you're using the annotation to store data, why not model the function on: `ActionsPageFeedItemWithIPFS.tsx` around line 51, which already implements much of the parsing/validation?